### PR TITLE
mmcsd: move block driver register to probe stage in order to read ext_csd reg

### PR DESCRIPTION
**Pull request by gneworld**
## Summary
  mmcsd: move block driver register to probe stage in order to read ext_csd reg
## Impact
  mmcsd
## Testing
 cortext M55

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX Requirements.**

**Missing Information:**

* **Summary:**
    *  A clear explanation of *why* this change is necessary is missing. What problem does moving the block driver registration solve? 
    * How does moving the registration to the probe stage enable reading the `ext_csd` register? Provide a technical explanation.
* **Impact:** 
    * The impact section is far too brief. Address *all* the bullet points (user impact, build impact, etc.) with "YES" or "NO" and provide details where applicable. Even if there's no impact, state "NO" explicitly for clarity.
* **Testing:**
    * **Insufficient Detail:** "cortext M55" is not enough information.  Specify:
        *  **Host OS:** (e.g., Linux Ubuntu 20.04)
        *  **Compiler Version:** (e.g., GCC 10.3.0)
        *  **NuttX Configuration:** (e.g., `configs/cortex-m55/nsh`)
    * **Missing Logs:** Provide actual testing logs from *before* and *after* the change to demonstrate the issue and the fix. 

**Recommendations:**

1. **Elaborate on the "Why":** Clearly state the problem that this change addresses. Why is reading the `ext_csd` register important, and why was it not possible before?
2. **Explain the Technical Details:**  Describe the code changes and how they enable reading the `ext_csd` register.
3. **Complete the Impact Section:**  Carefully consider and address all impact points, even if the answer is "NO".
4. **Provide Detailed Testing Information:** Include specific host and target details, along with actual testing logs demonstrating the problem and the solution.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@gneworld please fix(https://github.com/apache/nuttx/actions/runs/11084304188/job/30799485530?pr=13705):
```
mmcsd/mmcsd_sdio.c: In function 'mmcsd_slotinitialize':
Error: mmcsd/mmcsd_sdio.c:4367:47: error: 'devname' undeclared (first use in this function); did you mean 'rename'?
 4367 |   finfo("MMC: %s %" PRIu64 "KB %s %s mode\n", devname,
      |                                               ^~~~~~~

```

File: drivers/mmcsd/mmcsd_sdio.c [hunk 1]
```diff
@@ -4362,32 +4361,21 @@ int mmcsd_slotinitialize(int minor, FAR struct sdio_dev_s *dev)
...
-      goto errout_with_hwinit;
-    }
-
 #ifdef CONFIG_MMCSD_PROCFS
   mmcsd_initialize_procfs();
 #endif
 
+  /* Create a MMCSD device name */
+
+  snprintf(devname, 16, "/dev/mmcsd%d", minor);
```
Comment (pkarashchenko) [inline drivers/mmcsd/mmcsd_sdio.c:4370]:
```suggestion
  snprintf(devname, sizeof(devname), "/dev/mmcsd%d", minor);
```

Comment (gneworld) [reply drivers/mmcsd/mmcsd_sdio.c]: done!

**Comment by gneworld** [comment]
> @gneworld please fix(https://github.com/apache/nuttx/actions/runs/11084304188/job/30799485530?pr=13705):
> 
> ```
> mmcsd/mmcsd_sdio.c: In function 'mmcsd_slotinitialize':
> Error: mmcsd/mmcsd_sdio.c:4367:47: error: 'devname' undeclared (first use in this function); did you mean 'rename'?
>  4367 |   finfo("MMC: %s %" PRIu64 "KB %s %s mode\n", devname,
>       |                                               ^~~~~~~
> ```

@xiaoxiang781216  
done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**